pkg/ottl/contexts/ottlprofile/profile.go

```diff
@@ -0,0 +1,232 @@
+	pep := pathExpressionParser{telemetrySettings}
+	p, err := ottl.NewParser[TransformContext](
+		functions,
+		pep.parsePath,
+		telemetrySettings,
+	)
+	if err != nil {
+		return ottl.Parser[TransformContext]{}, err
+	}
+	for _, opt := range options {
+		opt(&p)
+	}
+	return p, nil
+}
+
+// EnablePathContextNames enables the support to path's context names on statements.
+// When this option is configured, all statement's paths must have a valid context prefix,
+// otherwise an error is reported.
+//
+// Experimental: *NOTE* this option is subject to change or removal in the future.
+func EnablePathContextNames() Option {
+	return func(p *ottl.Parser[TransformContext]) {
+		ottl.WithPathContextNames[TransformContext]([]string{
+			ctxprofile.Name,
+			ctxscope.LegacyName,
+			ctxresource.Name,
+		})(p)
+	}
+}
+
+type StatementSequenceOption func(*ottl.StatementSequence[TransformContext])
+
+func WithStatementSequenceErrorMode(errorMode ottl.ErrorMode) StatementSequenceOption {
+	return func(s *ottl.StatementSequence[TransformContext]) {
+		ottl.WithStatementSequenceErrorMode[TransformContext](errorMode)(s)
+	}
+}
+
+func NewStatementSequence(statements []*ottl.Statement[TransformContext], telemetrySettings component.TelemetrySettings, options ...StatementSequenceOption) ottl.StatementSequence[TransformContext] {
+	s := ottl.NewStatementSequence(statements, telemetrySettings)
+	for _, op := range options {
+		op(&s)
+	}
+	return s
+}
+
+type ConditionSequenceOption func(*ottl.ConditionSequence[TransformContext])
+
+func WithConditionSequenceErrorMode(errorMode ottl.ErrorMode) ConditionSequenceOption {
+	return func(c *ottl.ConditionSequence[TransformContext]) {
+		ottl.WithConditionSequenceErrorMode[TransformContext](errorMode)(c)
+	}
+}
+
+func NewConditionSequence(conditions []*ottl.Condition[TransformContext], telemetrySettings component.TelemetrySettings, options ...ConditionSequenceOption) ottl.ConditionSequence[TransformContext] {
+	c := ottl.NewConditionSequence(conditions, telemetrySettings)
+	for _, op := range options {
+		op(&c)
+	}
+	return c
+}
+
+type pathExpressionParser struct {
+	telemetrySettings component.TelemetrySettings
+}
+
+func (pep *pathExpressionParser) parsePath(path ottl.Path[TransformContext]) (ottl.GetSetter[TransformContext], error) {
+	if path == nil {
+		return nil, fmt.Errorf("path cannot be nil")
+	}
+	// Higher contexts parsing
+	if path.Context() != "" && path.Context() != ctxprofile.Name {
+		return pep.parseHigherContextPath(path.Context(), path)
+	}
+	// Backward compatibility with paths without context
+	if path.Context() == "" && (path.Name() == ctxresource.Name || path.Name() == ctxscope.LegacyName) {
+		return pep.parseHigherContextPath(path.Name(), path.Next())
+	}
+
+	switch path.Name() {
+	case "cache":
+		if path.Keys() == nil {
+			return accessCache(), nil
+		}
+		return accessCacheKey(path.Keys()), nil
```

Comment:
We had another refactor of the cache stuff, checkout `ottllogs` for an example.

Reply:
Thanks for the hint. Refactored like in ottllog/ottlmetric.